Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -505,6 +609,38 @@ public:
                     // shuffle
                     ShuffleFisherYates(indices_1, cell_start_1, cell_stop_1, engine);
                     ShuffleFisherYates(indices_2, cell_start_2, cell_stop_2, engine);
+                }
+            );
+
+            // Loop over independent particle pairs
+            amrex::ParallelForRNG( n_independent_pairs,
+                [=] AMREX_GPU_DEVICE (int i_coll, amrex::RandomEngine const& engine) noexcept
+                {
+                    // to avoid type mismatch errors
+                    auto ui_coll = (index_type)i_coll;
+
+                    // Use a bisection algorithm to find the index of the cell in which this pair is located
+                    int i_cell = amrex::bisect( p_coll_offsets, 0, n_cells, ui_coll );
```

Comment:
To fix `clang-tidy` error:
```suggestion
                    const int i_cell = amrex::bisect( p_coll_offsets, 0, n_cells, ui_coll );
```